app/models/book.rb

```diff
@@ -0,0 +1,20 @@
+class Book < ApplicationRecord
+  belongs_to :author
+
+  validates :name, presence: true
+  validates :price, presence: true
+
+  validate :validate_repeatancy, if: :check_record?
+
+  private
+
+  def validate_repeatancy
+    self.errors.add(:price, "Data already exists.")
+  end
+
+  def check_record?
+    @author = Author.find(self.author_id)
+    (!(@author.books.where(name: self.name).where(price: self.price).empty?))
+  end
```

Comment:
Instead of custom validation you can use 
validates :name, presence: true, unique: true
and it will work
Instead of finding all the records from DB and checking if it present or not